scripts/experiment_runs.py

```diff
@@ -104,7 +109,7 @@ def run_experiment(config: Config):
     best_prompt, best_score = best_prompt_callback.get_best_prompt()
```

Comment:
the optimize method returns the population of the last step. This is important, since the best performing one on the dev set, might not be the best one on the test set. Using a callback gives the user more flexibility to decide what they want to have returned (they might also create a custom one). Also: The best prompt might not be in the population of the last step. taking account of this, would overload the "responsiblity" of the optimizer classes.